Maple2.Server.Game/Manager/Items/InventoryManager.cs

```diff
@@ -172,6 +173,9 @@ public class InventoryManager {
 
             IList<(Item, int Added)> result = items.Add(add, true);
             if (result.Count == 0) {
+                if (mailIfFull) {
+                    MailItem(add);
```

Comment:
I feel like returning true makes more sense after mailing, do we need to return false to handle something?

Reply:
yes. I need a way to differentiate between if it failed to add to the inventory and mailed. if i had it set to false, it would return the same if it just mailed. Take for example line 247 in ItemBoxManager. I need to know if it failed so I can define the error to stop the box opening process later in the loop

Reply:
hmm, that's difficult because I feel like there needs to be a better way to differentiate between failure and mailed items. Maybe we should just leave `Add` as-is and expose `MailItem`. So usage would be something like:

```cs
if (!inventory.Add(item)) {
    inventory.MailItem(item);
}
```

Reply:
wouldn't this create the item twice?

Reply:
Yea good point, perhaps we can add a check for `OpenSlots <= 0` and then return early without creating item in DB. If an item cannot stack, and there's no slots then we know the add will fail so don't need to create in DB yet.